recipe/meta.yaml

```diff
@@ -39,25 +39,24 @@ requirements:
   host:
     - python
     - pip
-    - numpy
```

Comment:
I'm wondering if this was done to pin numpy at runtime because pythran compiles code at runtime... From what I see, Pythran knows how to convert some numpy functions to native code (https://pythran.readthedocs.io/en/latest/SUPPORT.html#numpy). There is an example at https://pythran.readthedocs.io/en/latest/MANUAL.html#:~:text=But%20scientific%20computing%20in%20Python%20usually%20means%20Numpy.%20Here%20is%20a%20well%2Dknown%20Numpy%20snippet%3A (open in Chrome).

Because of that, do we need to pin numpy like if it was used at build time?

Reply:
https://pythran.readthedocs.io/en/latest/MANUAL.html#integration-into-a-python-package also says:

>When distributing a Python application with Pythran modules, you can either:
>* use the module as a regular Python module (a .py file). After all, they are 100% Python compatible.
>* use Pythran as a Python to C++ transpiler and then integrate the generated C++ into your package build the same way as you would with other C++ code (see SciPy for an example of doing this with Meson).
>* Use Pythran’s PythranExtension in order to directly build extension modules as part of your build with setuptools.

Point 2 and 3 seems to support what I'm saying.

Reply:
Wouldn't that be in the package using pythran? Especially, from your first link: _Note that Pythran itself is a header-only library_

Reply:
Yeah, it's true that point 2 and 3 means that the user has to add numpy in their build deps.

Reply:
Yeah, I'm wondering why pythran is dependent on numpy anyway

Reply:
Probably easiest to take it out of host and see what happens..?

Reply:
Yeah the above seems to suggest, if your software uses numpy (or any other package that pythran can cross-compile bits of), then you should provide numpy at build time to get those cross-compilations. But not that pythran itself needs it at build time (of pythran)

Reply:
I did take it out of host, as per your suggestion.

Reply:
Yeah, that's fine. I was a little worried when I initially made the comment, but I was misunderstanding how it works.